### PR TITLE
Mini-PR: Minor style cleaning

### PR DESCRIPTION
**Pull request by ghugo83**
This is a minor PR to improve the style of https://github.com/cms-sw/cmssw/pull/32218, where I had not yet taken the time to clean.
- Add comments and use of std::find_if in  SimG4Core/Geometry/src/DD4hep_DDG4Builder.cc.
- It is not needed anymore to close geometry in DetectorDescription/DDCMS/plugins/dd4hep/DDDefinitions2Objects.cc.

@ianna @civanch @cvuosalo

### CODE REVIEW

File: SimG4Core/Geometry/src/DD4hep_DDG4Builder.cc [hunk 1]
```diff
@@ -51,27 +55,37 @@ G4VPhysicalVolume *DDG4Builder::BuildGeometry(SensitiveDetectorCatalog &catalog)
...
+    if (hasG4ReflectedVolume) {
+      // Add reflected sensitive detector to catalogue
       catalog.insert(
-          {sClassName.data(), sClassName.size()}, {sROUName.data(), sROUName.size()}, reflectedG4LogicalVolumeName);
-    }
+          {sClassName.data(), sClassName.size()}, {sROUName.data(), sROUName.size()}, sensitiveDetectorG4ReflectedName);
 
-    edm::LogVerbatim("SimG4CoreApplication")
-        << " DDG4SensitiveConverter: Sensitive " << fff << " Class Name " << sClassName << " ROU Name " << sROUName;
+      edm::LogVerbatim("SimG4CoreApplication")
```
Comment (civanch) [inline SimG4Core/Geometry/src/DD4hep_DDG4Builder.cc:85]: @ghugo83 , it may be too much to add the same printout as in line 67 ?

Comment (ghugo83) [reply SimG4Core/Geometry/src/DD4hep_DDG4Builder.cc:85]: @civanch This is the printout of the reflected sensors when they exist, while the printout at line 67 is the printout of the non-reflected sensors.

**Comment by civanch** [comment]
please test

File: DetectorDescription/DDCMS/plugins/dd4hep/DDDefinitions2Objects.cc [hunk 2]
```diff
@@ -2243,9 +2243,6 @@ static long load_dddefinition(Detector& det, xml_h element) {
       if (mfv1.isValid())
         wv.placeVolume(mfv1, 1);
 
-      // Can not deal with reflections without closed geometry
-      det.manager().CloseGeometry();
```
Comment (ianna) [inline DetectorDescription/DDCMS/plugins/dd4hep/DDDefinitions2Objects.cc:2247]: @ghugo83 - there is no harm in closing geometry earlier. DD4hep will close it any way. At least we'd know when we forbid any further modifications to geometry.

**Review by ianna — APPROVED** [review]
We ought to test if leaving geometry open is safe. Then ask for an option from DD4hep to keep it open. This will save us time on voxelization that we do not use :-)

**Comment by ianna** [comment]
+1

**Comment by ianna** [comment]
Hmm.. It's a unit test for reflected volume that fails:
```
world_volume/testLogicalParts:MotherOfAllBoxes_2/testLogicalParts:V_1101_1333
#80: V_1101_refl[1333] is a  == Solid not initialized
F

DDSolid.cppunit.cc:46:Assertion
Test name: testDDSolid::checkDDSolid
assertion failed
- Expression: title.compare(name) == 0

Failures !!!
Run: 1   Failure total: 1   Failures: 1   Errors: 0

---> test testDD4hepDDSolid had ERRORS
 
^^^^ End Test testDD4hepDDSolid ^^^^
```

File: DetectorDescription/DDCMS/plugins/dd4hep/DDDefinitions2Objects.cc (hunk 2 above)
Comment (ghugo83) [reply DetectorDescription/DDCMS/plugins/dd4hep/DDDefinitions2Objects.cc:2247]: @ianna It looks like we need to close the geometry.
When I revert this line, the tests pass.

**Comment by ghugo83** [comment]
Tests should pass now

**Comment by civanch** [comment]
please test

**Comment by civanch** [comment]
+1

**Comment by qliphy** [comment]
+1

**Pull request merged**